src/Common/src/System/Net/NetworkInformation/UnixCommandLinePing.cs

```diff
@@ -62,6 +64,22 @@ public static string ConstructCommandLine(int packetSize, string address, bool i
             // The ping utility is not flexible enough to specify an exact payload.
             // But we can at least send the right number of bytes.
 
+            if (ttl > 0)
+            {
+                if (!ipv4 && s_isOSX)
+                {
+                    // OSX uses -h to set hop limit for IPv6
+                    sb.Append(" -h ");
+                }
+                else
+                {
+                    // Linux uses -t ttl for both IPv4 & IPv6
```

Comment:
And macOS uses -t for ipv4?

Reply:
yes. It is TTL for IPv4 and Hop Limit for IPv6.